src/lib/NodeHighlighter.ts

```diff
@@ -0,0 +1,220 @@
+import { DataAttribute, dataToDatasetAttributeName, getDataAttributesFromEventPath } from '../utils/dataAttributes';
+import { EventManager } from './EventManager';
+import { HighlightRenderer, IRenderer } from './HighlightRenderer';
+import { IElementClickMessageData } from './IFrameCommunicator';
+
+export const IntersectionThreshold = 0.95;
+
+export enum NodeHighlighterEventType {
+  ElementClicked = 'kk:element:click',
+}
+
+export type NodeHighlighterMessagesMap = {
+  readonly [NodeHighlighterEventType.ElementClicked]: (data: Partial<IElementClickMessageData>) => void;
+};
+
+export class NodeHighlighter {
```

Comment:
I would expect highlighter to only highlight, but it also processes the click events which makes me think the `Renderer` is actually highlighter and this is more like `NodeSmartLinkProvider/Manager`